pytorch_lightning/trainer/training_io.py

```diff
@@ -388,6 +388,14 @@ def restore_training_state(self, checkpoint):
         self.global_step = checkpoint['global_step']
         self.current_epoch = checkpoint['epoch']
 
+        # Deals with peculiarity of different global step for odd vs even num_training_batches
+        if abs((self.global_step + 1) % self.num_training_batches) > 1:
```

Comment:
Is this known?

Reply:
the `abs` looks suspicious...

Reply:
I don't think it's strictly necessary, I was just being safe allowing a +/-1. 

The check should be `==0` if the global step matched for odd vs even number of training steps, so if that's worked out then we shouldn't need it. Unless it's intended, in which case I think just ``(self.global_step + 1) % self.num_training_batches`` is sufficient since global step matches num_training_batches in the even case. 

I can remove the abs for this PR if it's preferable, but I felt being within 1 batch of the expected number of steps was fine to not throw a warning.

Reply:
I've removed the `abs` and made sure it handled accumulated batches properly. Can you think of anything else that might change the global step? 

Currently the only test that throws this warning now is ``test_restore_models/test_dp_resume`` since this changes the percentage of train data used when resuming in a new trainer. Not much we can about that.